icees_api/features/sql.py

```diff
@@ -576,7 +567,28 @@ def select_feature_matrix(
         )
         conn.execute("DROP VIEW if exists tmp")
         conn.execute(view_query)
-        table_name = "tmp"
+        # return the view table name 'tmp' for subsequent operations
+        return 'tmp'
```

Comment:
So cohort_features and table_name are mutually exclusive? Both wouldn't get passed into this function? Because overriding the table_name seems a little counterintuitive. Maybe we add a check to make sure that both aren't specified?

Reply:
@maximusunc cohort_features and table_name input parameters are not mutually exclusive, but independent variables. cohort_features specifies query conditions for a specific cohort, e.g., Sex == 'Female" while table_name is the name of the table, e.g., patient, for the cohort_features or query condition to be applied to create a specific cohort. This create_cohort_view() function checks if cohort_features input parameter is not empty, it creates a VIEW for the cohort, and then return the view name as table name for applying subsequent query operations to return counts, frequencies, etc.. If cohort_features is empty, which means there is no subcohorts specified, then the input table_name will be returned, which allows all subsequent query operations to be applied directly on the initial patient table_name without applying any cohort query constraint. Hope this makes sense. Let me know if there is better way to handle this.

Reply:
From what you describe, it kind of sounds like they are mutually exclusive. If cohort_features is specified, then a new cohort VIEW will be created and be queried upon and I guess that is a "tmp" table? But if cohort_features is not specified and table_name is, then that table will be queried upon. It would be very confusing if I passed it the table_name "alien" but also a cohort_feature of  species == "Human". You would basically ignore my "alien" table.

Reply:
@maximusunc Oh, I see where the confusion comes from. In this case, the query ```species == "Human"``` is only applied to the table_name "alien``` so the table_name is a required variable which specified the table where cohort_features query is applied on. In your example, the created view would be empty since "alien" table is not supposed to include any human. So I don't think they are mutually exclusive, but rather cohort_features depends on table_name since it needs to be applied to table_name to create a view. Does it make sense? Did I miss anything? I can add comments to clarify this if that helps.

Reply:
Ahhh, I understand now. Thanks for the clarification! This looks good!